src/firefly/energy.py

```diff
@@ -1,16 +1,53 @@
 import logging
+import warnings
 
 import qtawesome as qta
 from bluesky_queueserver_api import BPlan
+from ophydregistry import ComponentNotFound
+from pydm.widgets.label import PyDMLabel
+from pydm.widgets.line_edit import PyDMLineEdit
 from qtpy import QtCore, QtWidgets
+from qtpy.QtWidgets import QDialogButtonBox, QFormLayout, QLineEdit, QVBoxLayout
 from xraydb.xraydb import XrayDB
 
 from firefly import display
-from haven import exceptions, load_config, registry
+from haven import load_config, registry
 
 log = logging.getLogger(__name__)
 
 
+class EnergyCalibrationDialog(QtWidgets.QDialog):
+    """A dialog box for calibrating the energy."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.setWindowTitle("Energy calibration")
+
+        self.layout = QVBoxLayout()
+        self.setLayout(self.layout)
+        # Widgets for inputting calibration parameters
+        self.form_layout = QFormLayout()
+        self.layout.addLayout(self.form_layout)
+        self.form_layout.addRow(
+            "Energy readback:", PyDMLabel(self, init_channel="haven://energy.readback")
+        )
+        self.form_layout.addRow(
+            "Energy setpoint:",
+            PyDMLineEdit(self, init_channel="haven://energy.setpoint"),
+        )
+        self.form_layout.addRow(
+            "Calibrated energy:",
```

Comment:
How to use "Calibrated energy", e.g. input standard Cu edge?

Reply:
Opened issue for this: https://github.com/spc-group/haven/issues/236